pkg/client/unversioned/clientcmd/api/types.go

```diff
@@ -56,8 +56,13 @@ type Cluster struct {
 	LocationOfOrigin string
 	// Server is the address of the kubernetes cluster (https://hostname:port).
 	Server string `json:"server"`
-	// APIVersion is the preferred api version for communicating with the kubernetes cluster (v1, v2, etc).
+	// DEPRECATED: APIVersion is the preferred api version for communicating with the kubernetes cluster (v1, v2, etc).
+	//  Because a cluster can run multiple API groups and potentially multiple versions of each, it no longer makes sense to specify
+	//  a single value for the cluster version. Deprecate replace it with PreferredGroupVersions. It will be ignored if it is present.
 	APIVersion string `json:"api-version,omitempty"`
+	// PreferredGroupVersions takes a string slice of multiple group/version pairs (separated by comma), and informs version negotiation.
+	// It contains preferred group/versions for communicating with the kubernetes cluster (g1/v1, g2/v2, etc).
```

Comment:
The test is for `""` group so the group part is missing; I'll add tests for group/version and make it more clear. `PreferredGroupVersions` be used later for creating client.Config's group/version (not yet implemented). Having plain version strings isn't enough since version doesn't make sense without its group and `Cluster` doesn't have group information.